Automattic/wp-calypso
# Checkout: Fix mobile issues

#### Changes proposed in this Pull Request

This PR makes the following tweaks for mobile checkout:
- Fix width of inactive pay button so that it doesn't overlap with the help button. 
- Hide payment method icons on small screens.

In addition to that, I had to refactor a number of Typescript errors that popped up.

**Before**
![image](https://user-images.githubusercontent.com/6981253/97582992-cbe90180-19cc-11eb-93b0-bbcdc98342d3.png)

![image](https://user-images.githubusercontent.com/6981253/97583092-ec18c080-19cc-11eb-8bfb-b41add78e71e.png)


**After**
![image](https://user-images.githubusercontent.com/6981253/97583181-081c6200-19cd-11eb-9a05-423c343ce996.png)

![image](https://user-images.githubusercontent.com/6981253/97583261-1a969b80-19cd-11eb-8ab6-93447ee87ae7.png)


#### Testing instructions

Visit checkout on a small screen and check:
- the help button doesn't overlap with an active and inactive pay button
- payment method icons in the payment method step are hidden

Then check things on desktop to make sure there are no regressions. Finally, complete a purchase to make sure nothing is broken.

Fixes https://github.com/Automattic/wp-calypso/issues/46572 along with https://github.com/Automattic/wp-calypso/pull/46780
